apollo-backend/src/main/java/io/logz/apollo/controllers/BlockerDefinitionController.java

```diff
@@ -172,6 +148,24 @@ public void updateBlockerDefinition(int id, String name, String environmentId, S
         assignJsonResponseToReq(req, HttpStatus.OK, blockerDefinition);
     }
 
+    private boolean isValid(String availability, Integer stackId, Integer environmentId, Integer serviceId) {
+        if (stackId != null && !stackId.equals("null")) {
+            if (environmentId != null || serviceId != null) {
```

Comment:
I can have environment stack and a service ID, or service stack and environment ID